File: 1 тиждень/credit/credit.c
#include <stdio.h>
#include <cs50.h>
long normlong(string abc);
int main(void)
{
    int s, p, count = 0;
    long t, ten=100000000000000, ten1=1000000000000000, ten2=1000000000000;
    long number = normlong("Namber: ");
    if((number>=340000000000000&&number<=349999999999999)||(number>=370000000000000&&number<=379999999999999))
    {
        for(int i=0;i<15;i++)
        {
            if(i%2==0)
            {
                s = number/ten;
                count+=s;
                t = number%ten;
                number = t;
                ten/=10;
            }
            else
            {
                s = number/ten;
                if(s<5)
                {
                    s*=2;
                    count+=s;
                    t = number%ten;
                    number = t;
                    ten/=10;
                }
                else
                {
                    s*=2;
                    p = s/10+s%10;
                    s = p;
                    count+=s;
                    t = number%ten;
                    number = t;
                    ten/=10;
                }   
            }        
        }
        if(count%10==0)
        {
            printf("AMEX\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }
    else if(number>=5100000000000000&&number<=5599999999999999)
    {
        for(int i=0;i<16;i++)
        {
            if(i%2==1)
            {
                s = number/ten1;
                count+=s;
                t = number%ten1;
                number = t;
                ten1/=10;
            }
            else
            {
                s = number/ten1;
                if(s<5)
                {
                    s*=2;
                    count+=s;
                    t = number%ten1;
                    number = t;
                    ten1/=10;
                }
                else
                {
                    s*=2;
                    p = s/10+s%10;
                    s = p;
                    count+=s;
                    t = number%ten1;
                    number = t;
                    ten1/=10;
                }   
            }        
        }
        if(count%10==0)
        {
            printf("MASTERCARD\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }
    else if(number>=4000000000000000&&number<=4999999999999999)
    {
        for(int i=0;i<16;i++)
        {
            if(i%2==1)
            {
                s = number/ten1;
                count+=s;
                t = number%ten1;
                number = t;
                ten1/=10;
            }
            else
            {
                s = number/ten1;
                if(s<5)
                {
                    s*=2;
                    count+=s;
                    t = number%ten1;
                    number = t;
                    ten1/=10;
                }
                else
                {
                    s*=2;
                    p = s/10+s%10;
                    s = p;
                    count+=s;
                    t = number%ten1;
                    number = t;
                    ten1/=10;
                }   
            }        
        }
        if(count%10==0)
        {
            printf("VISA\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }
    else if(number>=4000000000000&&number<=4999999999999)
    {
        for(int i=0;i<13;i++)
        {
            if(i%2==0)
            {
                s = number/ten2;
                count+=s;
                t = number%ten2;
                number = t;
                ten2/=10;
            }
            else
            {
                s = number/ten2;
                if(s<5)
                {
                    s*=2;
                    count+=s;
                    t = number%ten2;
                    number = t;
                    ten2/=10;
                }
                else
                {
                    s*=2;
                    p = s/10+s%10;
                    s = p;
                    count+=s;
                    t = number%ten2;
                    number = t;
                    ten2/=10;
                }   
            }        
        }
        if(count%10==0)
        {
            printf("VISA\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }
    else
    {
        printf("INVALID\n");
    }
}
long normlong(string abc)
{
    long m;
    do
    {
        m = get_long("%s", abc);
    }
    while(m<0);
    return m;
}

File: 6 тиждень/credit/credit.py
from cs50 import get_string
count = 0
k = 1
while k > 0:
    m = 0
    nomer = get_string("Number: ")
    for c in nomer:
        if ord(c) < 48 or ord(c) > 57:
            m += 1
    k = m
if len(nomer) == 15 and nomer[0] == "3" and (nomer[1] == "4" or nomer[1] == "7"):
    for d in range(len(nomer)):
        l = int(nomer[d])
        if d % 2 == 1:
            if l >= 5:
                l *= 2
                l = l % 10 + 1
                count += l
            else:
                l *= 2
                count += l
        else:
            count += l
    if count % 10 == 0:
        print("AMEX")
    else:
        print("INVALID")
elif len(nomer) == 16 and nomer[0] == "5" and (nomer[1] == "1" or nomer[1] == "2" or nomer[1] == "3" or nomer[1] == "4" or nomer[1] == "5"):
    for d in range(len(nomer)):
        l = int(nomer[d])
        if d % 2 == 1:
            count += l
        else:
            if l >= 5:
                l *= 2
                l = l % 10+1
                count += l
            else:
                l *= 2
                count += l
    if count % 10 == 0:
        print("MASTERCARD")
    else:
        print("INVALID")
elif len(nomer) == 16 and nomer[0] == "4":
    for d in range(len(nomer)):
        l = int(nomer[d])
        if d % 2 == 1:
            count += l
        else:
            if l >= 5:
                l *= 2
                l = l % 10+1
                count += l
            else:
                l *= 2
                count += l
    if count % 10 == 0:
        print("VISA")
    else:
        print("INVALID")
elif len(nomer) == 13 and nomer[0] == "4":
    for d in range(len(nomer)):
        l = int(nomer[d])
        if d % 2 == 1:
            if l >= 5:
                l *= 2
                l = l % 10 + 1
                count += l
            else:
                l *= 2
                count += l
        else:
            count += l
    print(count)
    if count % 10 == 0:
        print("VISA")
    else:
        print("INVALID")
else:
    print("INVALID")


